src/ui/Analytics/PendingSearchEvent.ts

```diff
@@ -47,6 +48,14 @@ export class PendingSearchEvent {
     return this.templateSearchEvent.customData;
   }
 
+  public addFacetsState(state: IAnalyticsDynamicFacetMeta[]) {
+    if (!this.facetsState) {
+      this.facetsState = [];
```

Comment:
in this case it would always sent en empty array for facetsState, I dont really want that, I only want it defined for those that use the DynamicFacets